ibm/service/kms/resource_ibm_kms_key.go

```diff
@@ -322,8 +322,11 @@ func populateKPClient(d *schema.ResourceData, meta interface{}, instanceID strin
 	if err != nil {
 		return nil, nil, err
 	}
+	var endpointType string
 
-	endpointType := d.Get("endpoint_type").(string)
```

Comment:
endpoint_type has default value "public" so we can use  d.Get no need for d.GetOk(...)